### DISCUSSION TITLE
Text Chat Event Hook + Client Text Chat Messanger

### DISCUSSION THREAD

**Discussion by DrDallas1** [SmartlyDressedGames/Unturned-3.x-Community#4332] [Q&A]
Hia everyone,

I recently ran into a problem while trying to create a system oriented around the client text chat messanger + text chat event hook.

I'm trying to make the process "hidden", so the player preferrably doesnt see anything in chat when the system goes through, but I can't figure out a way to make it invisible. I tried a command, which did work in singleplayer, but it failed in multiplayer (because it was picked up by rocket!). It also didnt seem to support certain unicode characters, so an "empty character" was bust. 

![image](https://github.com/SmartlyDressedGames/Unturned-3.x-Community/assets/66083394/8aba5649-f4e8-404a-a2d1-bd04ad8d56a4)

![image](https://github.com/SmartlyDressedGames/Unturned-3.x-Community/assets/66083394/b894c485-bbc0-49f3-b3b0-ea41a35f09dd)

![image](https://github.com/SmartlyDressedGames/Unturned-3.x-Community/assets/66083394/bc7cf231-a8f1-465f-aaaa-25f84cd2428d)

Ideas? As far as I see, there is absolutely no other way for me to remake this system pluginless. Might have to just settle for a local message in chat.

**Comment by SDGNelson** [comment]
IIRC plugins being able to handle those messages was one of the intentions: Your plugin can listen for `onCheckUnityEventPermissions` and set `shouldList` to false to hide it.
